.github/workflows/release-plz.yml

```diff
@@ -0,0 +1,45 @@
+name: Release-plz (main)
+
+permissions:
+  contents: read
+
+on:
+  workflow_call:
+
+jobs:
+  release-plz-release:
+    name: Release-plz release
+    runs-on: ubuntu-latest
+    if: ${{ github.repository_owner == '0xMiden' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: main
+      # Ensure the release tag refers to the latest commit on main.
+      # Compare the commit SHA that triggered the workflow with the HEAD of the branch we just
+      # checked out (main).
+      - name: Verify release was triggered from main HEAD
+        run: |
+          tag_sha="${{ github.sha }}"
+          main_sha="$(git rev-parse HEAD)"
+
+          echo "Tag points to:        $tag_sha"
+          echo "Current main HEAD is: $main_sha"
+
+          if [ "$tag_sha" != "$main_sha" ]; then
+            echo "::error::The release tag was not created from the latest commit on main. Aborting."
+            exit 1
+          fi
+          echo "Release tag matches main HEAD — continuing."
+      - name: Update Rust toolchain
+        run: |
+          rustup update --no-self-update
+      - name: Run release-plz
+        uses: release-plz/action@v0.5
+        with:
+          command: release
```

Comment:
We have a release-plz config file, see e.g. https://github.com/0xMiden/miden-base/blob/55c33c1afa3eed54af15c124d3c000b2bb927e92/.release-plz.toml

Any repo having a `[[package]]` section will need to define its own config file unfortunately

Reply:
The default should be from miden-node: https://github.com/0xMiden/miden-node/blob/de41f9c325f53f2457107294edadaba8c163cc01/.release-plz.toml